music/src/core/player.ts

```diff
@@ -201,20 +227,35 @@ export class Player {
     }
     return this.synths.get(instrument);
   }
+}
 
-  /**
-   * Stop playing the currently playing sequence right away.
-   */
-  stop() {
-    if (this.currentPart) {
-      this.currentPart.stop();
-      this.currentPart = null;
-    }
-    Tone.Transport.clear(this.scheduledStop);
-    this.scheduledStop = null;
+/**
+ * A `NoteSequence` player based on Tone.js that uses SoundFont samples.
+ */
+export class GMPlayer extends BasePlayer {
+  private soundfont: soundfont.SoundFont;
+
+  constructor(soundFontURL: string) {
+    super();
+    this.soundfont = new soundfont.SoundFont(soundFontURL);
   }
 
-  isPlaying() {
-    return !!this.currentPart;
+  async initialize() {
+    await this.soundfont.initialize();
+  }
+
+  async loadSamples(seq: INoteSequence): Promise<void> {
+    await this.soundfont.loadSamples(seq.notes.map((note) => ({
+                                                     pitch: note.pitch,
+                                                     velocity: note.velocity,
+                                                     program: note.program,
+                                                     isDrum: note.isDrum
+                                                   })));
+  }
+
+  protected playNote(time: number, note: NoteSequence.INote) {
```

Comment:
what happens if initialize hasn't been called?

Reply:
In order to match the behavior of the other Player, the implication is that playNote will immediately schedule the note and does not depend on disk/network calls.  The Instrument will throw an error if the sample hasn't already been loaded.  I'll add a comment.